pkg/cmd/server/start/start_node.go

```diff
@@ -256,7 +267,7 @@ func (o NodeOptions) RunNode() error {
 		return nil
 	}
 
-	if err := StartNode(*nodeConfig, o.NodeArgs.Components); err != nil {
+	if err := StartNode(*nodeConfig, o.NodeArgs.Components, o.NodeArgs.WriteFlagsOnly); err != nil {
```

Comment:
Rather than passing an "behave differently flag" to `StartNode`, you can detect and print out the options from a different method right above this.

Reply:
ok